web-api/src/storage/elastic.rs

```diff
@@ -359,65 +364,93 @@ impl storage::Document for Storage {
                 ]
             }))
             .await?;
-
-        if response.errors {
-            let failed_documents = response.items
-                .into_iter()
-                .filter_map(|mut response| {
-                    if let Some(response) = response.remove("index") {
-                        if !is_success_status(response.status, false) {
-                            error!(document_id=%response.id, error=%response.error, "Elastic failed to ingest document.");
-                            return Some(response.id.into());
+        let mut failed_documents = response
+            .errors
+            .then(|| {
+                response
+                    .items
+                    .into_iter()
+                    .filter_map(|mut response| {
+                        if let Some(response) = response.remove("index") {
+                            if !is_success_status(response.status, false) {
+                                error!(
+                                    document_id=%response.id,
+                                    error=%response.error,
+                                    "Elastic failed to ingest document.",
+                                );
+                                ids.remove(&response.id);
+                                return Some(response.id.into());
+                            }
+                        } else {
+                            error!("Bulk index request contains non index responses: {response:?}");
                         }
-                    } else {
-                        error!("Bulk index request contains non index responses: {response:?}");
-                    }
-                    None
-                })
-                .collect_vec();
+                        None
+                    })
+                    .collect_vec()
+            })
+            .unwrap_or_default();
+
+        if let Err(error) = self
+            .postgres
+            .insert_documents(&ids.into_iter().collect_vec(/* Itertools::chunks() is !Send */))
+            .await
+        {
+            if let InsertionError::PartialFailure {
+                failed_documents: fd,
+            } = error
+            {
+                failed_documents.extend(fd);
```

Comment:
:question: Should we try to delete the documents from elastic? We can have a service that cleans the state by checking which documents are on Postgres and removing them from elastic if not present.

Reply:
a cleanup service makes sense, should we create a new story for that?